### PR TITLE
Introduce link

### PR DESCRIPTION
**Pull request by andrewn617**
Implement a basic link. For now the link just takes a block (so it functions more like a `button`). It can only be rendered within a `para` otherwise it raises an exception. For now I just styled it as a `u`.

Still to do after lunch:
- add some tests
- make checking for the parent more general than just `is_a? Para`
- proper styling
- `click:` option for external urls
- `click:` option for internal routes

### CODE REVIEW

File: lib/scarpe/link.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    end
+
+    def render(parent)
+      raise "Links must be rendered with a para" unless parent.is_a? Para
+
+      @app.bind(function_name) do
+        @block&.call
+      end
+
+      html = HTML.render do |h|
```
Comment (gregoriokusowski) [inline lib/scarpe/link.rb:20]:
```suggestion
      HTML.render do |h|
```

File: lib/scarpe/para.rb [hunk 2]
```diff
@@ -7,11 +7,16 @@ def initialize(app, text)
     end
 
     def render
-      "<p id=#{object_id}>#{text.join}</p>"
+      "<p id=#{object_id}>#{render_text}</p>"
```
Comment (gregoriokusowski) [inline lib/scarpe/para.rb:10]: Should we move this to `HTML.render` now?

**Review by gregoriokusowski — COMMENTED** [review]
Adding a small suggestion to remove an extra var.

File: lib/scarpe/para.rb (hunk 2 above)
Comment (andrewn617) [reply lib/scarpe/para.rb]: Rebasing took care of this for me :D

**Pull request closed without merging**